### PR TITLE
Add more options on outputting time in console

### LINKED ISSUE #24944: Two more time units with hours and days
## Reason
We have users liking time in days such as `1.5d` better than `1d 12:00:00` in dtime. So it is desired to add few more options when formatting time.

## Design
We will just add two more options for the parameter `time_format` of `Console` in `MooseEnum`. Tests will be also added.

## Impact
New minor feature for improved console output.

### PR DESCRIPTION
**Pull request by YaqiWang**
Closes #24944.

### CODE REVIEW

File: framework/src/outputs/Console.C [hunk 1]
```diff
@@ -460,18 +460,26 @@ std::string
...
     if (_precision > 0)
       oss << std::setw(_precision) << std::setprecision(_precision) << std::setfill('0')
           << std::showpoint;
     if (_scientific_time)
       oss << std::scientific;
-    oss << t;
-    if (_time_format == "second")
-      oss << "s";
+
+    if (_time_format == "plain")
```
Comment (dschwen) [inline framework/src/outputs/Console.C:471]: I suggest using an enum class here and the `getEnum<...>()` call, like in [all other places where we frequently check enum values](https://github.com/search?q=repo%3Aidaholab%2Fmoose%20getEnum&type=code).

**Comment by YaqiWang** [comment]
More comments? Since @GiudGiud reviewed my previous PR on this, he might be interested in this change.

**Review by GiudGiud — APPROVED** [review]
no harm in this. I doubt anyone will want minutes or hours but time will tell

**Comment by GiudGiud** [comment]
Thanks for the contribution!

**Comment by GiudGiud** [comment]
@YaqiWang this (and the other PR) could go in the July newsletter btw

**Comment by YaqiWang** [comment]
Glad to add something in the newsletter. I have not contributed to the newsletter for a while. Ping me when it is time and please remind me the procedure.

**Comment by GiudGiud** [comment]
You just make a PR to change the newsletter file. It s created much ahead of time, and just advertised on the day off

You can wait until we make the final release of the webpage PR or do it at any time during the month

**Comment by YaqiWang** [comment]
Do I need to create an issue for the PR?

**Comment by GiudGiud** [comment]
nope. You can use the usual newsletter issue #11447
or your PR's issue

**Pull request merged**